### PR TITLE
Check python version against jar version

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
@cseed took a different approach, generating a python file which I import in hail initialization. This seemed like the only way to make things work in all the ways Python can be deployed (zip, egg, directory)

File: hail/generate-build-info.sh [hunk 1]
```diff
@@ -15,3 +16,4 @@ echo_build_properties() {
 mkdir -p src/main/resources/
 
 echo_build_properties $1 $2 > "src/main/resources/build-info.properties"
+python scripts/parse_version_info.py src/main/resources/build-info.properties python/hail/_generated_version_info.py
```
Comment (cseed) [inline hail/generate-build-info.sh]: Why not just create here python/hail/_generated_version_info.py the same way we do with build-info.properties?

**Comment by tpoterba** [comment]
yes, that's better. done.

File: hail/generate-build-info.sh [hunk 2]
```diff
@@ -2,16 +2,25 @@
...
+  echo branch=$BRANCH
+  echo date=$DATE
+  echo url=$URL
+  echo sparkVersion=$SPARK_VERSION
+  echo hailVersion=$HAIL_VERSION
 }
 
 mkdir -p src/main/resources/
 
 echo_build_properties $1 $2 > "src/main/resources/build-info.properties"
```
Comment (cseed) [inline hail/generate-build-info.sh]: Also, this echo function business is a bit unusual.  I'd just use the

```
cat <<DONE
blah blah blah
DONE
```

idiom.  But this is really an unrelated change.

Comment (tpoterba) [reply hail/generate-build-info.sh]: yeah, that's what I would have done too

**Comment by tpoterba** [comment]
uh, this worked locally -- what shell is the CI server running? Can I change "sh" to "bash"?
